code/cmdline/cmdline.cpp

```diff
@@ -1036,6 +1036,105 @@ char *cmdline_parm::str()
 	return args;
 }
 
+#ifdef SCP_UNIX
+// Return a vector with all filesystem names of "parent/dir" relative to parent.
+// dir must not contain a slash.
+static SCP_vector<SCP_string> unix_get_single_dir_names(SCP_string dir, SCP_string parent)
+{
+	DIR *dp;
+	if ((dp = opendir(parent.c_str())) == NULL) {
+		Error(LOCATION, "Can't open directory '%s' -- %d", parent.c_str(), errno);
```

Comment:
`Error` is a user-facing error which aborts execution. Either the error message has to be made more descriptive or the function should be altered to properly handle this error (e.g. by returning an empty vector).